src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -62,24 +63,37 @@ public GCSMultiBatchSink(GCSMultiBatchSinkConfig config) {
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    config.validate();
-    FileFormat format = config.getFormat();
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(collector);
+    FileFormat format = null;
+    try {
+      format = config.getFormat();
+    } catch (IllegalArgumentException e) {
+      // this failure must already be collected by collector. So it can be ignored
+    }
     // add schema as a macro since we don't know it until runtime
     PluginProperties formatProperties = PluginProperties.builder()
       .addAll(config.getProperties().getProperties())
       .add("schema", String.format("${%s}", SCHEMA_MACRO)).build();
-    OutputFormatProvider outputFormatProvider =
-      pipelineConfigurer.usePlugin(BatchSink.FORMAT_PLUGIN_TYPE, format.name().toLowerCase(),
-                                   FORMAT_PLUGIN_ID, formatProperties);
-    if (outputFormatProvider == null) {
-      throw new IllegalArgumentException(String.format("Could not find the '%s' output format plugin.",
-                                                       format.name().toLowerCase()));
+
+    if (format != null) {
```

Comment:
don't think this check is needed. Instead, if the format couldn't be fetched above, we can just throw the exception there since everything below it requires that format is not null.